ranger-discovery-bundle/src/main/java/io/appform/ranger/discovery/bundle/id/formatter/IdFormatters.java

```diff
@@ -31,4 +40,26 @@ public static IdFormatter base36() {
         return base36IdFormatter;
     }
 
+    /**
+     * Generate id by parsing given string
+     *
+     * @param idString String idString
+     * @return ID if it could be generated
+     */
+    public Optional<Id> parse(final String idString) {
```

Comment:
Remove this

Reply:
This will be used for parsing all IDs.

Reply:
This would be required. So there are two layers to parsing - 

1. To determine which formatter was used to generate this ID (in future plan is to add two digit unique identifier per formatter for this)
2. Once formatter is defined, individual formatters would be invoked to further parse all related metadata from the ID  


This method is only trying to identify  the formatter. Rest would be done by individual formatters. We can probably move this method to a new class (say IDParsers)